src/wallet/mod.rs

```diff
@@ -664,6 +664,40 @@ impl Wallet {
 		Ok(address_info.address.script_pubkey())
 	}
 
+	#[allow(deprecated)]
+	pub(crate) fn sign_owned_inputs(&self, unsigned_tx: Transaction) -> Result<Transaction, ()> {
+		let locked_wallet = self.inner.lock().unwrap();
+
+		let mut psbt = Psbt::from_unsigned_tx(unsigned_tx).map_err(|e| {
+			log_error!(self.logger, "Failed to construct PSBT: {}", e);
+		})?;
+		for (i, txin) in psbt.unsigned_tx.input.iter().enumerate() {
+			if let Some(utxo) = locked_wallet.get_utxo(txin.previous_output) {
+				psbt.inputs[i] = locked_wallet.get_psbt_input(utxo, None, true).map_err(|e| {
+					log_error!(self.logger, "Failed to construct PSBT input: {}", e);
+				})?;
+			}
+		}
+
+		let mut sign_options = SignOptions::default();
+		sign_options.trust_witness_utxo = true;
+
+		match locked_wallet.sign(&mut psbt, sign_options) {
+			Ok(finalized) => debug_assert!(!finalized),
+			Err(e) => {
+				log_error!(self.logger, "Failed to sign owned inputs: {}", e);
+				return Err(());
+			},
+		}
+
+		let mut tx = psbt.unsigned_tx;
```

Comment:
Just wondering, is there any reason we went this way instead of `extract_tx()`, as below? Do you think there would be any benefit in DRYing this up with `sign_psbt_inner`, or is it even preferable to keep them entirely separate?

Reply:
Unfortunately, `extract_tx` doesn't work because we can't sign the shared input (or our counterparty's inputs) here.